patches/minecraft/net/minecraft/client/renderer/entity/RenderItem.java.patch

```diff
@@ -9,7 +9,61 @@
          this.func_175041_b();
      }
  
-@@ -226,7 +226,7 @@
+@@ -153,7 +153,7 @@
+ 
+                 if (p_180454_1_.func_77962_s())
+                 {
+-                    this.func_180451_a(p_180454_2_);
++                    this.func_180451_a(p_180454_2_, p_180454_1_);
+                 }
+             }
+ 
+@@ -161,33 +161,18 @@
+         }
+     }
+ 
+-    private void func_180451_a(IBakedModel p_180451_1_)
++    private void func_180451_a(IBakedModel p_180451_1_, ItemStack stack)
+     {
+-        GlStateManager.func_179132_a(false);
```

Comment:
@RainWarrior Sorry I just want to make sure it's worth ripping out custom effect support as there's little to no reason to include it when it'll have to follow the vanilla glint's translation and scaling

I've only found [model type batching] (https://github.com/mezz/JustEnoughItems/blob/1.8.9/src/main/java/mezz/jei/gui/ingredients/GuiItemStackFastList.java#L110)... and then the overlay is run through afterwards which is the text information... the state is [still being flipped](https://github.com/mezz/JustEnoughItems/blob/1.8.9/src/main/java/mezz/jei/gui/ingredients/GuiItemStackFast.java#L170) during the model rendering for the effect passes since they're all handled in the one method... at least from what I've seen, aside from the batched lighting calls, which can be moved out from the passes, it is virtually identical to [my approach](https://github.com/MinecraftForge/MinecraftForge/pull/2458/files#diff-e3a7ac5de7307337f8f7a2fd797d506aR166)

Sorry @mezz I hope you don't mind me asking for some input on the matter from you

With JEI for supporting this you'd end up putting
```java
if (itemStack.hasEffect()) {
    EffectPassHandler handler = itemStack.getItem.getItemEffectRenderer(itemStack);
    if(handler.isEssential()){
        handler.bindTexture
        //run the passes
    }
```
In the renderEffect method to allow for the custom effects to be run through instead. Totally an optional thing to do since JEI already has its own included method that it uses for effect rendering.

The only time I can see that JEI would run into the patched calls with the EffectPassHandler support is included is when the it advocates to 'renderSlowly'(via vanilla)... unless I am missing something?